xray/httptrace_go17.go

```diff
@@ -112,11 +113,20 @@ func (xt *HTTPSubsegments) GotConn(info *httptrace.GotConnInfo, err error) {
 
 		if err == nil {
 			xt.reqCtx, _ = BeginSubsegment(xt.opCtx, "request")
+			xt.reqHeadersCtx, _ = BeginSubsegment(xt.reqCtx, "request-headers")
 		}
 
 	}
 }
 
+// WroteHeaders closes the request-headers subsegment if the HTTP operation
+// subsegment is still in progress.
+func (xt *HTTPSubsegments) WroteHeaders() {
+	if xt.reqHeadersCtx != nil && GetSegment(xt.opCtx).InProgress {
```

Comment:
Could we call [safeInProgress](https://github.com/aws/aws-xray-sdk-go/blob/master/xray/segment.go#L337) instead of directly access `InProgress` since it will cause unclose `in_progress` subsegment issue.

Reply:
Updated to use safeInProgress